### DISCUSSION TITLE
Self hosted docker Rustdesk server works only internal

### DISCUSSION THREAD

**Discussion by Azental** [rustdesk/rustdesk#1016] [Q&A]
Dear experts,

i got an issue, the server is working well on a docker container, if i setup my clients with the connection server to the local IP address, I’m able to connect to other devices in my network with RustDesk. Also, the client shows successful connected to our RustDesk docker Server.
If I try to use my external static IP address or a DNS forwarded to my IP address, I´m not able to connect to any clients local / external.
The client tries to connect and doesn’t reach the destination, after a few tries the software tells the “Deadline has elapsed”.
Our firewall got the open ports that’s needed: 21115 – 21119 TCP + 21116 UDP.
Do you have any Idea why the connection to a client is not working?

**Comment by rustdesk** [comment]
https://github.com/rustdesk/rustdesk-server/issues/24

**Reply by Azental** [reply]
Hi, 
I’m sorry, but I think that’s not the problem that i have.
I use already NAT and the clients will goes "green" like connected / ready, so I’m sure that the connection to the Server is working fine.
